lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1142,16 +1352,32 @@ void HdVP2Material::_UpdateShaderInstance(const HdMaterialNetwork& mat)
         HdVP2RenderDelegate::sProfilerCategory, MProfiler::kColorD_L2, "UpdateShaderInstance");
 
     for (const HdMaterialNode& node : mat.nodes) {
-        // Find the simplified path for the authored node path from the map which has been created
-        // when applying VP2-specific fixes.
-        const auto it = _nodePathMap.find(node.path);
-        if (it == _nodePathMap.end()) {
-            continue;
-        }
+        MString nodeName = "";
+#ifdef WANT_MATERIALX_BUILD
+        if (_isMtlx) {
+            nodeName += node.path.GetName().c_str();
+            if (node.path == _surfaceShaderId) {
+                nodeName = "";
+            } else {
+                nodeName += "_";
+            }
+        } else {
+#endif
+            // Find the simplified path for the authored node path from the map which has been
+            // created when applying VP2-specific fixes.
+            const auto it = _nodePathMap.find(node.path);
+            if (it == _nodePathMap.end()) {
+                continue;
+            }
 
-        // The simplified path has only one token which is the node name.
-        const SdfPath& nodePath = it->second;
-        const MString  nodeName(nodePath != _surfaceShaderId ? nodePath.GetText() : "");
+            // The simplified path has only one token which is the node name.
+            const SdfPath& nodePath = it->second;
+            if (nodePath != _surfaceShaderId) {
+                nodeName = nodePath.GetText();
+            }
+#ifdef WANT_MATERIALX_BUILD
```

Comment:
Rather than have an extra #ifdef just to end the scope you could move the open brace outside the above #ifdef and have that code in a scope.